csm/core/controllers/rgw/s3/base.py

```diff
@@ -16,10 +16,25 @@
 from contextlib import contextmanager
 from csm.core.controllers.view import CsmView, CsmHttpException
 from csm.core.services.rgw.s3.utils import S3ServiceError
+from marshmallow import Schema, ValidationError, validates_schema
 
 S3_SERVICE_ERROR = 0x3000
 
+class S3BaseSchema(Schema):
+    """
+    Base Class for S3 Schema Validation.
+    invalidate_empty_values : method with marshmallo schema validates_schema decorator
+    """
+    @classmethod
+    @validates_schema
+    def invalidate_empty_values(data, **kwargs):
+        """Method invalidates the empty strings."""
+        for key, value in data.items():
+            if value is not None and not str(value).strip():
+                raise ValidationError(f"{key}: Can not be empty")
+
```

Comment:
please check after the movement of this logic to parent class, are validations working for user APIs and Key APIs too.

Reply:
Done